ideascube/library/migrations/0008_library_stock.py

```diff
@@ -23,16 +27,45 @@ def migrate_data(apps, schema_editor):
             subtitle=old_book.subtitle, publisher=old_book.publisher,
             section=old_book.section, lang=old_book.lang, cover=old_book.cover,
             tags=old_book.tags)
-        new_book.save()
+        old_specimen = None
+        new_bookspecimens = []
 
         for old_bookspecimen in old_book.specimens.all():
-            new_bookspecimen = BookSpecimen(
-                barcode=old_bookspecimen.serial, item=new_book, count=1,
-                comments=old_bookspecimen.remarks,
-                # And now the stuff which hasn't changed
-                location=old_bookspecimen.location, file=old_bookspecimen.file)
-            new_bookspecimen.save()
+            if old_bookspecimen.serial in specimens:
+                # If "old serial" equals "new barcode" then we are dealing with
+                # the same specimen
+                old_specimen = specimens.pop(old_bookspecimen.serial)
+
+            if old_specimen is not None:
+                # Either we just got old_specimen in the previous if condition,
+                # or we got it on a previous iteration
+
+                if new_book.id is None:
+                    new_book.stockitem_ptr = old_specimen.item
+                    new_book.id = old_specimen.item.id
```

Comment:
Hum, I understand why you do this two lines. You make the Book inherit from the StockItem you've just found from the old_specimen.

But, when you create the new Book (l.22), django should have create a (temporary) StockItem has the Book inherit from it. So the given `name` and `description` are stored in a this (temporary) StockItem. If you change the stockitem here, the `name` and `description` will(/may?) be lost.